<?xml version="1.0" encoding="UTF-8"?>
<testng-results skipped="1" failed="4" ignored="0" total="23" passed="18">
  <reporter-output>
    <line>
      <![CDATA[Clicking on login button By.linkText: Log in]]>
    </line>
    <line>
      <![CDATA[Clicking on login button By.linkText: Log in]]>
    </line>
  </reporter-output>
  <suite name="Surefire suite" duration-ms="63847" started-at="2023-01-29T17:37:41 GMT" finished-at="2023-01-29T17:38:45 GMT">
    <groups>
      <group name="regression">
        <method signature="TestNGGrouping.loginTest2()[pri:0, instance:com.nopcommerce.demo.testng.TestNGGrouping@6a1aab78]" name="loginTest2" class="com.nopcommerce.demo.testng.TestNGGrouping"/>
      </group> <!-- regression -->
      <group name="smoke">
        <method signature="TestNGGrouping.loginTest1()[pri:0, instance:com.nopcommerce.demo.testng.TestNGGrouping@6a1aab78]" name="loginTest1" class="com.nopcommerce.demo.testng.TestNGGrouping"/>
      </group> <!-- smoke -->
      <group name="sanity">
        <method signature="TestNGGrouping.loginTest3()[pri:0, instance:com.nopcommerce.demo.testng.TestNGGrouping@6a1aab78]" name="loginTest3" class="com.nopcommerce.demo.testng.TestNGGrouping"/>
      </group> <!-- sanity -->
    </groups>
    <test name="Surefire test" duration-ms="63847" started-at="2023-01-29T17:37:41 GMT" finished-at="2023-01-29T17:38:45 GMT">
      <class name="com.nopcommerce.demo.testng.TestNGPriority">
        <test-method status="PASS" signature="test()[pri:0, instance:com.nopcommerce.demo.testng.TestNGPriority@757942a1]" name="test" duration-ms="0" started-at="2023-01-29T17:37:42 GMT" finished-at="2023-01-29T17:37:42 GMT">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- test -->
        <test-method status="PASS" signature="test1()[pri:0, instance:com.nopcommerce.demo.testng.TestNGPriority@757942a1]" name="test1" duration-ms="1" started-at="2023-01-29T17:37:42 GMT" finished-at="2023-01-29T17:37:42 GMT">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- test1 -->
        <test-method status="PASS" signature="test2()[pri:0, instance:com.nopcommerce.demo.testng.TestNGPriority@757942a1]" name="test2" duration-ms="1" started-at="2023-01-29T17:37:42 GMT" finished-at="2023-01-29T17:37:42 GMT">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- test2 -->
      </class> <!-- com.nopcommerce.demo.testng.TestNGPriority -->
      <class name="com.nopcommerce.demo.testng.TestNGRetryFailedDemo">
        <test-method status="PASS" signature="test1()[pri:0, instance:com.nopcommerce.demo.testng.TestNGRetryFailedDemo@4a87761d]" name="test1" duration-ms="1" started-at="2023-01-29T17:37:42 GMT" finished-at="2023-01-29T17:37:42 GMT">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- test1 -->
        <test-method status="PASS" signature="test2()[pri:0, instance:com.nopcommerce.demo.testng.TestNGRetryFailedDemo@4a87761d]" name="test2" duration-ms="0" started-at="2023-01-29T17:37:42 GMT" finished-at="2023-01-29T17:37:42 GMT">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- test2 -->
        <test-method status="PASS" signature="test3()[pri:0, instance:com.nopcommerce.demo.testng.TestNGRetryFailedDemo@4a87761d]" name="test3" duration-ms="0" started-at="2023-01-29T17:37:42 GMT" finished-at="2023-01-29T17:37:42 GMT">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- test3 -->
      </class> <!-- com.nopcommerce.demo.testng.TestNGRetryFailedDemo -->
      <class name="com.nopcommerce.demo.testng.TestNGSoftAssert">
        <test-method status="FAIL" signature="testSum()[pri:0, instance:com.nopcommerce.demo.testng.TestNGSoftAssert@66d1af89]" name="testSum" duration-ms="4" started-at="2023-01-29T17:37:42 GMT" finished-at="2023-01-29T17:37:42 GMT">
          <exception class="java.lang.AssertionError">
            <message>
              <![CDATA[expected [true] but found [false]]]>
            </message>
            <full-stacktrace>
              <![CDATA[java.lang.AssertionError: expected [true] but found [false]
at org.testng.Assert.fail(Assert.java:110)
at org.testng.Assert.failNotEquals(Assert.java:1413)
at org.testng.Assert.assertTrue(Assert.java:56)
at org.testng.Assert.assertTrue(Assert.java:66)
at com.nopcommerce.demo.testng.TestNGSoftAssert.testSum(TestNGSoftAssert.java:18)
at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
at java.lang.reflect.Method.invoke(Method.java:498)
at org.testng.internal.invokers.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:135)
at org.testng.internal.invokers.TestInvoker.invokeMethod(TestInvoker.java:673)
at org.testng.internal.invokers.TestInvoker.invokeTestMethod(TestInvoker.java:220)
at org.testng.internal.invokers.MethodRunner.runInSequence(MethodRunner.java:50)
at org.testng.internal.invokers.TestInvoker$MethodInvocationAgent.invoke(TestInvoker.java:945)
at org.testng.internal.invokers.TestInvoker.invokeTestMethods(TestInvoker.java:193)
at org.testng.internal.invokers.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:146)
at org.testng.internal.invokers.TestMethodWorker.run(TestMethodWorker.java:128)
at java.util.ArrayList.forEach(ArrayList.java:1249)
at org.testng.TestRunner.privateRun(TestRunner.java:808)
at org.testng.TestRunner.run(TestRunner.java:603)
at org.testng.SuiteRunner.runTest(SuiteRunner.java:429)
at org.testng.SuiteRunner.runSequentially(SuiteRunner.java:423)
at org.testng.SuiteRunner.privateRun(SuiteRunner.java:383)
at org.testng.SuiteRunner.run(SuiteRunner.java:326)
at org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:52)
at org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:95)
at org.testng.TestNG.runSuitesSequentially(TestNG.java:1249)
at org.testng.TestNG.runSuitesLocally(TestNG.java:1169)
at org.testng.TestNG.runSuites(TestNG.java:1092)
at org.testng.TestNG.run(TestNG.java:1060)
at org.apache.maven.surefire.testng.TestNGExecutor.run(TestNGExecutor.java:135)
at org.apache.maven.surefire.testng.TestNGDirectoryTestSuite.executeMulti(TestNGDirectoryTestSuite.java:193)
at org.apache.maven.surefire.testng.TestNGDirectoryTestSuite.execute(TestNGDirectoryTestSuite.java:94)
at org.apache.maven.surefire.testng.TestNGProvider.invoke(TestNGProvider.java:146)
at org.apache.maven.surefire.booter.ForkedBooter.invokeProviderInSameClassLoader(ForkedBooter.java:384)
at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:345)
at org.apache.maven.surefire.booter.ForkedBooter.execute(ForkedBooter.java:126)
at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:418)
]]>
            </full-stacktrace>
          </exception> <!-- java.lang.AssertionError -->
          <reporter-output>
          </reporter-output>
        </test-method> <!-- testSum -->
        <test-method status="FAIL" signature="testSum1()[pri:0, instance:com.nopcommerce.demo.testng.TestNGSoftAssert@66d1af89]" name="testSum1" duration-ms="20" started-at="2023-01-29T17:37:42 GMT" finished-at="2023-01-29T17:37:42 GMT">
          <exception class="java.lang.AssertionError">
            <message>
              <![CDATA[The following asserts failed:
Result not equal to 40 expected [true] but found [false]]]>
            </message>
            <full-stacktrace>
              <![CDATA[java.lang.AssertionError: The following asserts failed:
Result not equal to 40 expected [true] but found [false]
at org.testng.asserts.SoftAssert.assertAll(SoftAssert.java:46)
at org.testng.asserts.SoftAssert.assertAll(SoftAssert.java:30)
at com.nopcommerce.demo.testng.TestNGSoftAssert.testSum1(TestNGSoftAssert.java:35)
at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
at java.lang.reflect.Method.invoke(Method.java:498)
at org.testng.internal.invokers.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:135)
at org.testng.internal.invokers.TestInvoker.invokeMethod(TestInvoker.java:673)
at org.testng.internal.invokers.TestInvoker.invokeTestMethod(TestInvoker.java:220)
at org.testng.internal.invokers.MethodRunner.runInSequence(MethodRunner.java:50)
at org.testng.internal.invokers.TestInvoker$MethodInvocationAgent.invoke(TestInvoker.java:945)
at org.testng.internal.invokers.TestInvoker.invokeTestMethods(TestInvoker.java:193)
at org.testng.internal.invokers.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:146)
at org.testng.internal.invokers.TestMethodWorker.run(TestMethodWorker.java:128)
at java.util.ArrayList.forEach(ArrayList.java:1249)
at org.testng.TestRunner.privateRun(TestRunner.java:808)
at org.testng.TestRunner.run(TestRunner.java:603)
at org.testng.SuiteRunner.runTest(SuiteRunner.java:429)
at org.testng.SuiteRunner.runSequentially(SuiteRunner.java:423)
at org.testng.SuiteRunner.privateRun(SuiteRunner.java:383)
at org.testng.SuiteRunner.run(SuiteRunner.java:326)
at org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:52)
at org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:95)
at org.testng.TestNG.runSuitesSequentially(TestNG.java:1249)
at org.testng.TestNG.runSuitesLocally(TestNG.java:1169)
at org.testng.TestNG.runSuites(TestNG.java:1092)
at org.testng.TestNG.run(TestNG.java:1060)
at org.apache.maven.surefire.testng.TestNGExecutor.run(TestNGExecutor.java:135)
at org.apache.maven.surefire.testng.TestNGDirectoryTestSuite.executeMulti(TestNGDirectoryTestSuite.java:193)
at org.apache.maven.surefire.testng.TestNGDirectoryTestSuite.execute(TestNGDirectoryTestSuite.java:94)
at org.apache.maven.surefire.testng.TestNGProvider.invoke(TestNGProvider.java:146)
at org.apache.maven.surefire.booter.ForkedBooter.invokeProviderInSameClassLoader(ForkedBooter.java:384)
at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:345)
at org.apache.maven.surefire.booter.ForkedBooter.execute(ForkedBooter.java:126)
at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:418)
]]>
            </full-stacktrace>
          </exception> <!-- java.lang.AssertionError -->
          <reporter-output>
          </reporter-output>
        </test-method> <!-- testSum1 -->
      </class> <!-- com.nopcommerce.demo.testng.TestNGSoftAssert -->
      <class name="com.nopcommerce.demo.testng.TestNGDemo">
        <test-method status="PASS" signature="beforeTest()[pri:0, instance:com.nopcommerce.demo.testng.TestNGDemo@74ad1f1f]" name="beforeTest" is-config="true" duration-ms="47" started-at="2023-01-29T17:37:41 GMT" finished-at="2023-01-29T17:37:41 GMT">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- beforeTest -->
        <test-method status="PASS" signature="beforeMethod()[pri:0, instance:com.nopcommerce.demo.testng.TestNGDemo@74ad1f1f]" name="beforeMethod" is-config="true" duration-ms="0" started-at="2023-01-29T17:37:41 GMT" finished-at="2023-01-29T17:37:41 GMT">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- beforeMethod -->
        <test-method status="PASS" signature="test1()[pri:0, instance:com.nopcommerce.demo.testng.TestNGDemo@74ad1f1f]" name="test1" duration-ms="1" started-at="2023-01-29T17:37:41 GMT" finished-at="2023-01-29T17:37:41 GMT">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- test1 -->
        <test-method status="PASS" signature="afterMethod()[pri:0, instance:com.nopcommerce.demo.testng.TestNGDemo@74ad1f1f]" name="afterMethod" is-config="true" duration-ms="0" started-at="2023-01-29T17:37:41 GMT" finished-at="2023-01-29T17:37:41 GMT">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- afterMethod -->
        <test-method status="PASS" signature="test2()[pri:0, instance:com.nopcommerce.demo.testng.TestNGDemo@74ad1f1f]" name="test2" duration-ms="1" started-at="2023-01-29T17:37:41 GMT" finished-at="2023-01-29T17:37:41 GMT">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- test2 -->
        <test-method status="PASS" signature="beforeMethod()[pri:0, instance:com.nopcommerce.demo.testng.TestNGDemo@74ad1f1f]" name="beforeMethod" is-config="true" duration-ms="0" started-at="2023-01-29T17:37:41 GMT" finished-at="2023-01-29T17:37:41 GMT">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- beforeMethod -->
        <test-method status="PASS" signature="afterMethod()[pri:0, instance:com.nopcommerce.demo.testng.TestNGDemo@74ad1f1f]" name="afterMethod" is-config="true" duration-ms="0" started-at="2023-01-29T17:37:41 GMT" finished-at="2023-01-29T17:37:41 GMT">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- afterMethod -->
        <test-method status="PASS" signature="afterTest()[pri:0, instance:com.nopcommerce.demo.testng.TestNGDemo@74ad1f1f]" name="afterTest" is-config="true" duration-ms="0" started-at="2023-01-29T17:38:45 GMT" finished-at="2023-01-29T17:38:45 GMT">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- afterTest -->
      </class> <!-- com.nopcommerce.demo.testng.TestNGDemo -->
      <class name="com.nopcommerce.demo.testsuite.LoginPageTest">
        <test-method status="PASS" signature="setUp()[pri:0, instance:com.nopcommerce.demo.testsuite.LoginPageTest@2a70a3d8]" name="setUp" is-config="true" duration-ms="39963" started-at="2023-01-29T17:37:42 GMT" finished-at="2023-01-29T17:38:22 GMT">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- setUp -->
        <test-method status="PASS" signature="verifyErrorMessageWithInvalidCredentials()[pri:0, instance:com.nopcommerce.demo.testsuite.LoginPageTest@2a70a3d8]" name="verifyErrorMessageWithInvalidCredentials" duration-ms="7592" started-at="2023-01-29T17:38:22 GMT" finished-at="2023-01-29T17:38:29 GMT">
          <reporter-output>
            <line>
              <![CDATA[Clicking on login button By.linkText: Log in]]>
            </line>
          </reporter-output>
        </test-method> <!-- verifyErrorMessageWithInvalidCredentials -->
        <test-method status="PASS" signature="tearDown()[pri:0, instance:com.nopcommerce.demo.testsuite.LoginPageTest@2a70a3d8]" name="tearDown" is-config="true" duration-ms="3977" started-at="2023-01-29T17:38:29 GMT" finished-at="2023-01-29T17:38:33 GMT">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- tearDown -->
        <test-method status="PASS" signature="setUp()[pri:0, instance:com.nopcommerce.demo.testsuite.LoginPageTest@2a70a3d8]" name="setUp" is-config="true" duration-ms="8392" started-at="2023-01-29T17:38:33 GMT" finished-at="2023-01-29T17:38:42 GMT">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- setUp -->
        <test-method status="PASS" signature="verifyUserShouldNavigateToLoginPage()[pri:0, instance:com.nopcommerce.demo.testsuite.LoginPageTest@2a70a3d8]" name="verifyUserShouldNavigateToLoginPage" duration-ms="1536" started-at="2023-01-29T17:38:42 GMT" finished-at="2023-01-29T17:38:43 GMT">
          <reporter-output>
            <line>
              <![CDATA[Clicking on login button By.linkText: Log in]]>
            </line>
          </reporter-output>
        </test-method> <!-- verifyUserShouldNavigateToLoginPage -->
        <test-method status="PASS" signature="tearDown()[pri:0, instance:com.nopcommerce.demo.testsuite.LoginPageTest@2a70a3d8]" name="tearDown" is-config="true" duration-ms="1941" started-at="2023-01-29T17:38:43 GMT" finished-at="2023-01-29T17:38:45 GMT">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- tearDown -->
      </class> <!-- com.nopcommerce.demo.testsuite.LoginPageTest -->
      <class name="com.nopcommerce.demo.testng.TestNGParameterDemo">
        <test-method status="FAIL" signature="test(java.lang.String,java.lang.String)[pri:0, instance:com.nopcommerce.demo.testng.TestNGParameterDemo@69b0fd6f]" name="test" duration-ms="0" started-at="2023-01-29T17:37:42 GMT" finished-at="2023-01-29T17:37:42 GMT">
          <exception class="org.testng.TestNGException">
            <message>
              <![CDATA[
Cannot inject @Test annotated Method [test] with [class java.lang.String, class java.lang.String].
For more information on native dependency injection please refer to https://testng.org/doc/documentation-main.html#native-dependency-injection]]>
            </message>
            <full-stacktrace>
              <![CDATA[org.testng.TestNGException: 
Cannot inject @Test annotated Method [test] with [class java.lang.String, class java.lang.String].
For more information on native dependency injection please refer to https://testng.org/doc/documentation-main.html#native-dependency-injection
at org.testng.internal.Parameters.checkParameterTypes(Parameters.java:436)
at org.testng.internal.Parameters.createParametersForMethod(Parameters.java:350)
at org.testng.internal.Parameters.createParameters(Parameters.java:669)
at org.testng.internal.Parameters.handleParameters(Parameters.java:856)
at org.testng.internal.Parameters.handleParameters(Parameters.java:704)
at org.testng.internal.invokers.ParameterHandler.handleParameters(ParameterHandler.java:70)
at org.testng.internal.invokers.ParameterHandler.createParameters(ParameterHandler.java:49)
at org.testng.internal.invokers.TestInvoker$MethodInvocationAgent.invoke(TestInvoker.java:904)
at org.testng.internal.invokers.TestInvoker.invokeTestMethods(TestInvoker.java:193)
at org.testng.internal.invokers.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:146)
at org.testng.internal.invokers.TestMethodWorker.run(TestMethodWorker.java:128)
at java.util.ArrayList.forEach(ArrayList.java:1249)
at org.testng.TestRunner.privateRun(TestRunner.java:808)
at org.testng.TestRunner.run(TestRunner.java:603)
at org.testng.SuiteRunner.runTest(SuiteRunner.java:429)
at org.testng.SuiteRunner.runSequentially(SuiteRunner.java:423)
at org.testng.SuiteRunner.privateRun(SuiteRunner.java:383)
at org.testng.SuiteRunner.run(SuiteRunner.java:326)
at org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:52)
at org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:95)
at org.testng.TestNG.runSuitesSequentially(TestNG.java:1249)
at org.testng.TestNG.runSuitesLocally(TestNG.java:1169)
at org.testng.TestNG.runSuites(TestNG.java:1092)
at org.testng.TestNG.run(TestNG.java:1060)
at org.apache.maven.surefire.testng.TestNGExecutor.run(TestNGExecutor.java:135)
at org.apache.maven.surefire.testng.TestNGDirectoryTestSuite.executeMulti(TestNGDirectoryTestSuite.java:193)
at org.apache.maven.surefire.testng.TestNGDirectoryTestSuite.execute(TestNGDirectoryTestSuite.java:94)
at org.apache.maven.surefire.testng.TestNGProvider.invoke(TestNGProvider.java:146)
at org.apache.maven.surefire.booter.ForkedBooter.invokeProviderInSameClassLoader(ForkedBooter.java:384)
at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:345)
at org.apache.maven.surefire.booter.ForkedBooter.execute(ForkedBooter.java:126)
at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:418)
]]>
            </full-stacktrace>
          </exception> <!-- org.testng.TestNGException -->
          <reporter-output>
          </reporter-output>
        </test-method> <!-- test -->
      </class> <!-- com.nopcommerce.demo.testng.TestNGParameterDemo -->
      <class name="com.nopcommerce.demo.testng.TestNGGrouping">
        <test-method status="PASS" signature="computerTest1()[pri:0, instance:com.nopcommerce.demo.testng.TestNGGrouping@6a1aab78]" name="computerTest1" duration-ms="2" started-at="2023-01-29T17:37:42 GMT" finished-at="2023-01-29T17:37:42 GMT">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- computerTest1 -->
        <test-method status="PASS" signature="computerTest2()[pri:0, instance:com.nopcommerce.demo.testng.TestNGGrouping@6a1aab78]" name="computerTest2" duration-ms="0" started-at="2023-01-29T17:37:42 GMT" finished-at="2023-01-29T17:37:42 GMT">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- computerTest2 -->
        <test-method status="PASS" signature="loginTest1()[pri:0, instance:com.nopcommerce.demo.testng.TestNGGrouping@6a1aab78]" name="loginTest1" duration-ms="3" started-at="2023-01-29T17:37:42 GMT" finished-at="2023-01-29T17:37:42 GMT">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- loginTest1 -->
        <test-method status="PASS" signature="loginTest2()[pri:0, instance:com.nopcommerce.demo.testng.TestNGGrouping@6a1aab78]" name="loginTest2" duration-ms="8" started-at="2023-01-29T17:37:42 GMT" finished-at="2023-01-29T17:37:42 GMT">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- loginTest2 -->
        <test-method status="PASS" signature="loginTest3()[pri:0, instance:com.nopcommerce.demo.testng.TestNGGrouping@6a1aab78]" name="loginTest3" duration-ms="2" started-at="2023-01-29T17:37:42 GMT" finished-at="2023-01-29T17:37:42 GMT">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- loginTest3 -->
        <test-method status="PASS" signature="loginTest4()[pri:0, instance:com.nopcommerce.demo.testng.TestNGGrouping@6a1aab78]" name="loginTest4" duration-ms="0" started-at="2023-01-29T17:37:42 GMT" finished-at="2023-01-29T17:37:42 GMT">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- loginTest4 -->
        <test-method status="PASS" signature="regressionTest1()[pri:0, instance:com.nopcommerce.demo.testng.TestNGGrouping@6a1aab78]" name="regressionTest1" duration-ms="1" started-at="2023-01-29T17:37:42 GMT" finished-at="2023-01-29T17:37:42 GMT">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- regressionTest1 -->
      </class> <!-- com.nopcommerce.demo.testng.TestNGGrouping -->
      <class name="com.nopcommerce.demo.testng.TestNGListenersDemo">
        <test-method status="PASS" signature="test1()[pri:0, instance:com.nopcommerce.demo.testng.TestNGListenersDemo@462d5aee]" name="test1" duration-ms="0" started-at="2023-01-29T17:37:42 GMT" finished-at="2023-01-29T17:37:42 GMT">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- test1 -->
        <test-method status="FAIL" signature="test2()[pri:0, instance:com.nopcommerce.demo.testng.TestNGListenersDemo@462d5aee]" name="test2" duration-ms="21" started-at="2023-01-29T17:37:42 GMT" finished-at="2023-01-29T17:37:42 GMT">
          <exception class="java.lang.AssertionError">
            <message>
              <![CDATA[null]]>
            </message>
            <full-stacktrace>
              <![CDATA[java.lang.AssertionError: null
at org.testng.Assert.fail(Assert.java:110)
at org.testng.Assert.fail(Assert.java:115)
at com.nopcommerce.demo.testng.TestNGListenersDemo.test2(TestNGListenersDemo.java:21)
at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
at java.lang.reflect.Method.invoke(Method.java:498)
at org.testng.internal.invokers.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:135)
at org.testng.internal.invokers.TestInvoker.invokeMethod(TestInvoker.java:673)
at org.testng.internal.invokers.TestInvoker.invokeTestMethod(TestInvoker.java:220)
at org.testng.internal.invokers.MethodRunner.runInSequence(MethodRunner.java:50)
at org.testng.internal.invokers.TestInvoker$MethodInvocationAgent.invoke(TestInvoker.java:945)
at org.testng.internal.invokers.TestInvoker.invokeTestMethods(TestInvoker.java:193)
at org.testng.internal.invokers.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:146)
at org.testng.internal.invokers.TestMethodWorker.run(TestMethodWorker.java:128)
at java.util.ArrayList.forEach(ArrayList.java:1249)
at org.testng.TestRunner.privateRun(TestRunner.java:808)
at org.testng.TestRunner.run(TestRunner.java:603)
at org.testng.SuiteRunner.runTest(SuiteRunner.java:429)
at org.testng.SuiteRunner.runSequentially(SuiteRunner.java:423)
at org.testng.SuiteRunner.privateRun(SuiteRunner.java:383)
at org.testng.SuiteRunner.run(SuiteRunner.java:326)
at org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:52)
at org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:95)
at org.testng.TestNG.runSuitesSequentially(TestNG.java:1249)
at org.testng.TestNG.runSuitesLocally(TestNG.java:1169)
at org.testng.TestNG.runSuites(TestNG.java:1092)
at org.testng.TestNG.run(TestNG.java:1060)
at org.apache.maven.surefire.testng.TestNGExecutor.run(TestNGExecutor.java:135)
at org.apache.maven.surefire.testng.TestNGDirectoryTestSuite.executeMulti(TestNGDirectoryTestSuite.java:193)
at org.apache.maven.surefire.testng.TestNGDirectoryTestSuite.execute(TestNGDirectoryTestSuite.java:94)
at org.apache.maven.surefire.testng.TestNGProvider.invoke(TestNGProvider.java:146)
at org.apache.maven.surefire.booter.ForkedBooter.invokeProviderInSameClassLoader(ForkedBooter.java:384)
at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:345)
at org.apache.maven.surefire.booter.ForkedBooter.execute(ForkedBooter.java:126)
at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:418)
]]>
            </full-stacktrace>
          </exception> <!-- java.lang.AssertionError -->
          <reporter-output>
          </reporter-output>
        </test-method> <!-- test2 -->
        <test-method status="SKIP" signature="test3()[pri:0, instance:com.nopcommerce.demo.testng.TestNGListenersDemo@462d5aee]" name="test3" duration-ms="0" started-at="2023-01-29T17:37:42 GMT" finished-at="2023-01-29T17:37:42 GMT">
          <exception class="org.testng.SkipException">
            <message>
              <![CDATA[Test Skipped]]>
            </message>
            <full-stacktrace>
              <![CDATA[org.testng.SkipException: Test Skipped
at com.nopcommerce.demo.testng.TestNGListenersDemo.test3(TestNGListenersDemo.java:27)
at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
at java.lang.reflect.Method.invoke(Method.java:498)
at org.testng.internal.invokers.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:135)
at org.testng.internal.invokers.TestInvoker.invokeMethod(TestInvoker.java:673)
at org.testng.internal.invokers.TestInvoker.invokeTestMethod(TestInvoker.java:220)
at org.testng.internal.invokers.MethodRunner.runInSequence(MethodRunner.java:50)
at org.testng.internal.invokers.TestInvoker$MethodInvocationAgent.invoke(TestInvoker.java:945)
at org.testng.internal.invokers.TestInvoker.invokeTestMethods(TestInvoker.java:193)
at org.testng.internal.invokers.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:146)
at org.testng.internal.invokers.TestMethodWorker.run(TestMethodWorker.java:128)
at java.util.ArrayList.forEach(ArrayList.java:1249)
at org.testng.TestRunner.privateRun(TestRunner.java:808)
at org.testng.TestRunner.run(TestRunner.java:603)
at org.testng.SuiteRunner.runTest(SuiteRunner.java:429)
at org.testng.SuiteRunner.runSequentially(SuiteRunner.java:423)
at org.testng.SuiteRunner.privateRun(SuiteRunner.java:383)
at org.testng.SuiteRunner.run(SuiteRunner.java:326)
at org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:52)
at org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:95)
at org.testng.TestNG.runSuitesSequentially(TestNG.java:1249)
at org.testng.TestNG.runSuitesLocally(TestNG.java:1169)
at org.testng.TestNG.runSuites(TestNG.java:1092)
at org.testng.TestNG.run(TestNG.java:1060)
at org.apache.maven.surefire.testng.TestNGExecutor.run(TestNGExecutor.java:135)
at org.apache.maven.surefire.testng.TestNGDirectoryTestSuite.executeMulti(TestNGDirectoryTestSuite.java:193)
at org.apache.maven.surefire.testng.TestNGDirectoryTestSuite.execute(TestNGDirectoryTestSuite.java:94)
at org.apache.maven.surefire.testng.TestNGProvider.invoke(TestNGProvider.java:146)
at org.apache.maven.surefire.booter.ForkedBooter.invokeProviderInSameClassLoader(ForkedBooter.java:384)
at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:345)
at org.apache.maven.surefire.booter.ForkedBooter.execute(ForkedBooter.java:126)
at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:418)
]]>
            </full-stacktrace>
          </exception> <!-- org.testng.SkipException -->
          <reporter-output>
          </reporter-output>
        </test-method> <!-- test3 -->
      </class> <!-- com.nopcommerce.demo.testng.TestNGListenersDemo -->
    </test> <!-- Surefire test -->
  </suite> <!-- Surefire suite -->
</testng-results>
